include/drivers/espi_saf.h

```diff
@@ -0,0 +1,522 @@
+ *
+ * @retval true If eSPI SAF is ready.
+ * @retval false otherwise.
+ */
+__syscall bool espi_saf_get_channel_status(struct device *dev);
+
+static inline bool z_impl_espi_saf_get_channel_status(struct device *dev)
+{
+	const struct espi_saf_driver_api *api =
+		(const struct espi_saf_driver_api *)dev->api;
+
+	return api->get_channel_status(dev);
+}
+
+/**
+ * @brief Sends a read request packet for slave attached flash.
+ *
+ * This routines provides an interface to send a request to read the flash
+ * component shared between the eSPI master and eSPI slaves.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pckt Adddress of the representation of read flash transaction.
+ *
+ * @retval -ENOTSUP eSPI flash logical channel transactions not supported.
+ * @retval -EBUSY eSPI flash channel is not ready or disabled by master.
+ * @retval -EIO General input / output error, failed request to master.
+ */
+__syscall int espi_saf_flash_read(struct device *dev,
+				  struct espi_saf_packet *pckt);
+
+static inline int z_impl_espi_saf_flash_read(struct device *dev,
+					     struct espi_saf_packet *pckt)
+{
+	const struct espi_saf_driver_api *api =
+		(const struct espi_saf_driver_api *)dev->api;
+
+	if (!api->flash_read) {
+		return -ENOTSUP;
+	}
+
+	return api->flash_read(dev, pckt);
+}
+
+/**
+ * @brief Sends a write request packet for slave attached flash.
+ *
+ * This routines provides an interface to send a request to write to the flash
+ * components shared between the eSPI master and eSPI slaves.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pckt Address of the representation of write flash transaction.
+ *
+ * @retval -ENOTSUP eSPI flash logical channel transactions not supported.
+ * @retval -EBUSY eSPI flash channel is not ready or disabled by master.
+ * @retval -EIO General input / output error, failed request to master.
+ */
+__syscall int espi_saf_flash_write(struct device *dev,
+				   struct espi_saf_packet *pckt);
+
+static inline int z_impl_espi_saf_flash_write(struct device *dev,
+					      struct espi_saf_packet *pckt)
+{
+	const struct espi_saf_driver_api *api =
+		(const struct espi_saf_driver_api *)dev->api;
+
+	if (!api->flash_write) {
+		return -ENOTSUP;
+	}
+
+	return api->flash_write(dev, pckt);
+}
+
+/**
+ * @brief Sends a write request packet for slave attached flash.
+ *
+ * This routines provides an interface to send a request to write to the flash
+ * components shared between the eSPI master and eSPI slaves.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pckt Address of the representation of erase flash transaction.
+ *
+ * @retval -ENOTSUP eSPI flash logical channel transactions not supported.
+ * @retval -EBUSY eSPI flash channel is not ready or disabled by master.
+ * @retval -EIO General input / output error, failed request to master.
+ */
+__syscall int espi_saf_flash_erase(struct device *dev,
```

Comment:
where are the verification functions for this and the other system calls that are added in this PR?

Reply:
Is it acceptable to add verification tests in this PR?

Reply:
from my perspective it's mandatory.

Reply:
either that, or drop the `__syscall`.

Reply:
Ok. Making sure the test code is still working after pulling the latest samples/drivers/espi changes.

Reply:
Pushed rebase containing SAF tests added to samples/drivers/espi. This rebase defaults samples/drivers/espi to its default settings (no SAF). Edit the *.conf files and enable QMSPI and SAF to build the application with SAF support.